pkg/halstead/halstead.go

```diff
@@ -0,0 +1,363 @@
+	info.addToken(token.ELLIPSIS)
+}
+
+func (info *HalsteadInfo) AddEmptyStmt(node *ast.EmptyStmt) {
+	if !node.Implicit{
+		info.addToken(token.SEMICOLON)
+	}
+}
+
+func (info *HalsteadInfo) AddExprStmt(node *ast.ExprStmt) {
+	// Nothing to count
+	return
+}
+
+func (info *HalsteadInfo) AddField(node *ast.Field) {
+	// Nothing to count
+	return
+}
+
+func (info *HalsteadInfo) AddFieldList(node *ast.FieldList) {
+	if node.Opening.IsValid() {
+		info.addToken(token.LPAREN)
+	}
+}
+
+func (info *HalsteadInfo) AddFile(node *ast.File) {
+	info.addToken(token.PACKAGE)
+}
+
+func (info *HalsteadInfo) AddForStmt(node *ast.ForStmt) {
+	info.addToken(token.FOR)
+}
+
+func (info *HalsteadInfo) AddFuncDecl(node *ast.FuncDecl) {
+	// Composite type only, nothing
+	return
+}
+
+func (info *HalsteadInfo) AddFuncLit(node *ast.FuncLit) {
+	// Composite type only, nothing
+	return
+}
+
+func (info *HalsteadInfo) AddFuncType(node *ast.FuncType) {
+	info.addToken(token.FUNC)
+}
+
+func (info *HalsteadInfo) AddGenDecl(node *ast.GenDecl) {
+	info.addToken(node.Tok)
+	if node.Lparen.IsValid() {
+		info.addToken(token.LPAREN)
+	}
+}
+
+func (info *HalsteadInfo) AddGoStmt(node *ast.GoStmt) {
+	info.addToken(token.GO)
+}
+
+func (info *HalsteadInfo) AddIdent(node *ast.Ident) {
+	info.operands[node.Name] += 1
+}
+
+func (info *HalsteadInfo) AddIfStmt(node *ast.IfStmt) {
+	info.addToken(token.IF)
+}
+
+func (info *HalsteadInfo) AddImportSpec(node *ast.ImportSpec) {
+	// Composite type only, nothing
+	return
+}
+
+func (info *HalsteadInfo) AddIncDecStmt(node *ast.IncDecStmt) {
+	info.addToken(node.Tok)
+}
+
+func (info *HalsteadInfo) AddIndexExpr(node *ast.IndexExpr) {
+	if node.Lbrack.IsValid() {
+		info.addToken(token.LBRACK)
+	}
+}
+
+func (info *HalsteadInfo) AddInterfaceType(node *ast.InterfaceType) {
+	info.addToken(token.INTERFACE)
+}
+
+func (info *HalsteadInfo) AddKeyValueExpr(node *ast.KeyValueExpr) {
+	info.addToken(token.COLON)
+}
+
+func (info *HalsteadInfo) AddLabeledStmt(node *ast.LabeledStmt) {
+	info.addToken(token.COLON)
+}
+
+func (info *HalsteadInfo) AddMapType(node *ast.MapType) {
+	info.addToken(token.MAP)
+}
+
+func (info *HalsteadInfo) AddPackage(node *ast.Package) {
+	// name should be handled by ident, so ignore is as apparently it is 
+	// not a particular part of code but rather abstract entity (set of files?).
+	return
+}
+
+func (info *HalsteadInfo) AddParenExpr(node *ast.ParenExpr) {
+	info.addToken(token.LPAREN)
+}
+
+func (info *HalsteadInfo) AddRangeStmt(node *ast.RangeStmt) {
+	// for should be already handled by `AddForStmt`
+	info.addToken(node.Tok)
+	info.addToken(token.RANGE)
+}
+
+func (info *HalsteadInfo) AddReturnStmt(node *ast.ReturnStmt) {
+	info.addToken(token.RETURN)
+}
+
+func (info *HalsteadInfo) AddSelectStmt(node *ast.SelectStmt) {
+	info.addToken(token.SELECT)
+}
+
+func (info *HalsteadInfo) AddSelectorExpr(node *ast.SelectorExpr) {
+	info.addToken(token.PERIOD)
+	return
```

Comment:
No need to do explicit returns at the end of function

Reply:
solved in https://github.com/bragov4ik/go-kys/pull/13/commits/eb1dd136033e1864abf86d39912ec873f8277698